src/Identifier/LdapIdentifier.php

```diff
@@ -92,18 +118,63 @@ public function __construct(array $config = [])
      */
     protected function _checkLdapConfig()
     {
-        if (!isset($this->_config['bindDN'])) {
-            throw new RuntimeException('Config `bindDN` is not set.');
-        }
-        if (!is_callable($this->_config['bindDN'])) {
-            throw new InvalidArgumentException(sprintf(
-                'The `bindDN` config is not a callable. Got `%s` instead.',
-                gettype($this->_config['bindDN'])
-            ));
-        }
         if (!isset($this->_config['host'])) {
             throw new RuntimeException('Config `host` is not set.');
         }
+
+        if (!isset($this->_config['bindDN']) && (!isset($this->_config['filter']))) {
+            throw new RuntimeException('Config `bindDN` is not set.');
+        }
+
+        if (isset($this->_config['bindDN'])) {
+            if (!is_callable($this->_config['bindDN'])) {
+                throw new InvalidArgumentException(sprintf(
+                    'The `bindDN` config is not a callable. Got `%s` instead.',
+                    gettype($this->_config['bindDN'])
+                ));
+            }
+        } else {
+            if (!isset($this->_config['filter'])) {
+                throw new RuntimeException('Config `filter` is not set.');
+            }
+
+            if (!is_callable($this->_config['filter'])) {
+                throw new InvalidArgumentException(sprintf(
+                    'The `filter` config is not a callable. Got `%s` instead.',
+                    gettype($this->_config['filter'])
+                ));
+            }
+
+            if (!isset($this->_config['lookupBindPassword'])) {
+                throw new RuntimeException('Config `lookupBindPassword` is not set.');
+            }
+
+            if (!isset($this->_config['lookupBindPassword'])) {
+                throw new RuntimeException('Config `lookupBindPassword` is not set.');
+            }
+
+            if (!isset($this->_config['baseDN'])) {
+                throw new RuntimeException('Config `baseDN` is not set.');
+            }
+
+            if (isset($this->_config['updateLocalIdentity'])) {
+                if (!is_callable($this->_config['updateLocalIdentity'])) {
+                    throw new InvalidArgumentException(sprintf(
+                        'The `updateLocalIdentity` config is not a callable. Got `%s` instead.',
+                        gettype($this->_config['updateLocalIdentity'])
+                    ));
+                }
+            }
+
+            if (isset($this->_config['createLocalIdentityIfMissing'])) {
+                if (!is_callable($this->_config['createLocalIdentityIfMissing'])) {
```

Comment:
Should we check for a `Closure` instead of a callable? The callable type includes a string and array types that can be problematic to safely handle.